lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -148,11 +151,50 @@ function convertMessageToCtc(message, examples = {}) {
 
   _processPlaceholderDirectIcu(ctc, examples);
 
-  _ctcSanityChecks(ctc);
+  _ctcValidityChecks(ctc);
 
   return ctc;
 }
 
+/**
+ * Do some basic checks on an lhl message to confirm that it is valid. Future
+ * lhl regression catching should go here.
+ *
+ * @param {string} lhlMessage
+ */
+function _lhlValidityChecks(lhlMessage) {
+  let parsedMessage;
+  try {
+    parsedMessage = MessageParser.parse(lhlMessage);
+  } catch (err) {
+    if (err.name !== 'SyntaxError') throw err;
+    // Improve the intl-messageformat-parser syntax error output.
```

Comment:
for some reason the regex `intl-messageformat-parser` uses to pull out bad syntax for error messages usually only pulls out part of it (sometimes even just the first letter of an invalid word). It's possible this has been fixed in newer versions of the library, but they've changed their dependencies and how they handle number formatting, so this doesn't seem like the time to do that update. In the meantime, we'll just do our best with what they give us.